### ISSUE TITLE
Troubleshooting YOLOv8 Detection Training: Unable to Improve mAP50-95 Values Despite Various Adjustments

### ISSUE THREAD

**Issue by ibrahimciftci1** [ultralytics/ultralytics#9549]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm trying to conduct a detection training with YOLOv8, but the mAP50-95 value doesn't exceed 0.25-0.35. I've tried experimenting with S, M, L models instead of YOLOv8x, and I've also made changes with the IOU and confidence values. Additionally, I've tried adjusting the image size to 960x1280 from 640. I've experimented with batch sizes of 4, 8, and 16, but still couldn't achieve satisfactory results. I even attempted to reset everything by setting augment=False, hsv_h=0, hsv_s=0, hsv_v=0, degrees=0.0, translate=0, scale=0, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0, mosaic=0, and mixup=0.0. Despite trying mosaic=0, I'm unsure if it's still working or how to ensure it's not being used. Or where I might be making mistakes in the training process. I've also conducted numerous training attempts with default settings.

**Comment by glenn-jocher** [comment]
@ibrahimciftci1 hello there! 👋 It sounds like you've been trying a lot of adjustments to improve your model's mAP50-95 scores. Here are a few suggestions that might help:

1. **Data Quality:** Ensure your dataset has high-quality, well-labeled images. More diverse and larger datasets often help improve model performance.

2. **Learning Rate Schedule:** Experiment with different learning rate schedules. Sometimes starting with a higher learning rate and gradually decreasing it (lr scheduler) can lead to better convergence.

3. **Model Complexity:** While experimenting with S, M, L, and X models is great, ensure the model complexity matches your dataset size. Larger models require more data to generalize well.

4. **Augmentation:** Augmentations can be quite beneficial. Instead of disabling them, try using different combinations. For mosaic, to confirm it's disabled, ensure `mosaic=0.0` is properly set in your data YAML file or as an override parameter.

5. **Pretrained Weights:** Starting your training from pretrained weights (if not already doing so) can also give your model a good initial boost.

6. **Validation Set:** Use a representative validation set to accurately gauge model performance during training. This might help identify if the model is overfitting or underfitting.

Could you also share some details about your dataset? Sometimes, specific characteristics of the dataset require tailored strategies. 😊

Here's a simple code example to ensure you're setting the mosaic correctly:

```yaml
# In your dataset YAML file
mosaic: 0.0  # This disables mosaic augmentation
```

Or directly in your training command:

```bash
yolo train data=your_dataset.yaml model=yolov8n.pt mosaic=0.0
```

Keep experimenting, and don't hesitate to reach out for more assistance!

**Comment by ibrahimciftci1** [comment]
Thank you for your prompt response. You requested information about my dataset. I'm attempting to detect masses in a mammography dataset. The images are of size 2800x3518, and I have a total of 1029 images. I've allocated 10% for testing, 10% for validation, and 80% for training. When reviewing the results after training, upon examining val_batch_labels and val_batch_pred images, I noticed instances where multiple masses are detected in images where only one mass should be present, or conversely, there are cases where no mass is detected in images where there should be one.

**Comment by glenn-jocher** [comment]
@ibrahimciftci1, thanks for sharing details about your dataset. 📊 It seems like you're dealing with a relatively small and highly specialized dataset. In such cases, overfitting or under-detection could occur due to the model's complexity relative to the amount of data. Here are a few quick suggestions:

- **Augmentation:** Given the high resolution of your images, using augmentations that simulate zooming in or out, like scale and crop, might help the model learn from various perspectives and scales.

- **Image Size during Training:** Although your images are large, training with significantly smaller sizes could cause loss of detail. Experiment with closer-to-original sizes, if memory permits, or consider a step-wise reduction strategy.

- **Advanced Techniques:** Look into advanced detection techniques tailored for small object detection or few-shot learning, which could potentially increase detection performance on such datasets.

- **Annotation Review:** Double-check annotations for consistency and correctness, as errors here could directly affect detection quality.

For dealing with multiple detections where there should only be one, playing with `--iou-thres` for Non-Maximum Suppression during validation might help in reducing multiple detections for a single object:

```bash
yolo val data=your_dataset.yaml model=yolov8n.pt iou-thres=0.5
```

Adjusting the confidence threshold (`--conf-thres`) might help with the cases of missed detections:

```bash
yolo val data=your_dataset.yaml model=yolov8n.pt conf-thres=0.4
```

Remember, finding the right balance requires experimentation. Keep fine-tuning, and you should see improvements! 😊
